### PR TITLE
create home page

### LINKED ISSUE #43: create home page
## Description
Develop a home page that will be the first page for users after the login page, the home page will join the challenges-calendar, the challenges-completion and the user-level components in the same page.

image

## Motivation
To provide users with a clear and engaging way to know what they can do in the platform, including some CTA buttons and useful information for the end user.

## Useful Links
[Figma Design](https://www.figma.com/design/xUlM0BIpE50NX1YItvSdGu/Escopo-projeto?node-id=295-3925&t=1O3g48kyWmYmTRcs-4)

## Acceptance Criteria
- [ ] Add a route to the home page
- [ ] Join challenges-calendar, challenges-completion and user-level components to the page
- [ ] Create basic integrations between components, using [props drilling](https://www.alura.com.br/artigos/prop-drilling-no-react-js) if needed
- [ ] Add a image at the bottom of the page
- [ ] Integrate the sidebar menu in the main layout

### PR DESCRIPTION
**Pull request by Patolinow**
## Description
- Resolves #43

This PR introduces the following page:
- [home page](https://github.com/Super-Patos-Hackaton/frontend/blob/feat/create-home-page/src/app/(dashboard)/dashboard/page.tsx)
image

It also includes changes on several dependency components

## Todo
- [ ] Added a route to the home page
- [ ] Joined challenges-calendar, challenges-completion and user-level components to the page
- [ ] Adjusted sizes and colors from components
- [ ] Integrate the sidebar menu in the main layout

### CODE REVIEW

File: src/app/(dashboard)/dashboard/page.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import ChallengesCalendar from '@/components/challenges-calendar';
+import ChallengesCompletion from '@/components/challenges-completion';
+import UserLevel from '@/components/user-level';
+
+const Home = () => {
+  return (
+    <main className='mx-auto mt-[-48px] flex h-full max-w-7xl flex-col justify-center px-6'>
```
Comment (whyleonardo) [inline src/app/(dashboard)/dashboard/page.tsx]: This negative margin is necessary?

File: src/components/ui/widget.tsx [hunk 2]
```diff
@@ -2,10 +2,15 @@ import { ReactNode } from 'react';
...
 interface WidgetProps {
   children: ReactNode;
+  className?: string;
 }
 
-const Widget = ({ children }: WidgetProps) => {
-  return <div className='w-[576px] rounded-xl bg-secondary'>{children}</div>;
+const Widget = ({ children, className }: WidgetProps) => {
+  return (
+    <div className={'w-[576px] rounded-xl bg-secondary px-6 py-1 ' + className}>
```
Comment (whyleonardo) [inline src/components/ui/widget.tsx]: Use the helper function `cn()` to concat tailwind classes.

1. Import `cn` from `@/lib/utils`
2. Use:
 ```html
<div {className={cn('w-[576px] rounded-xl bg-secondary px-6 py-1', className)} />
 ```

This combine all possible classes who are passed in props and merge it without conflits.

**Review by whyleonardo — CHANGES_REQUESTED** [review]
Just some little changes to make

File: src/components/ui/widget.tsx (hunk 2 above)
Comment (Patolinow) [reply src/components/ui/widget.tsx]: fixed at the commit:
- [fix: fixed classname concat](https://github.com/Super-Patos-Hackaton/frontend/commit/85bc4f994d3b198346a0fcc182cb62c1c175d81c)

File: src/app/(dashboard)/dashboard/page.tsx (hunk 1 above)
Comment (Patolinow) [reply src/app/(dashboard)/dashboard/page.tsx]: fixed at the commit 
- [fix: replaced negative margin-top with positive padding-bottom](https://github.com/Super-Patos-Hackaton/frontend/commit/28ff626214a13fde1994482f0ad2002d10c24d41)

**Review by whyleonardo — APPROVED** [review]

**Comment by whyleonardo** [comment]
lgtm

**Pull request merged**